request-example.go

```diff
@@ -69,6 +75,48 @@ func (fs *root) Filewrite(r *Request) (io.WriterAt, error) {
 	return file.WriterAt()
 }
 
+func (fs *root) CheckFile(r *Request) (CheckFileResponse, error) {
+	response := CheckFileResponse{}
+	requestedAlgos := r.CheckFile.GetHashAlgoAsList()
+	var h hash.Hash
+	for _, algo := range requestedAlgos {
+		if algo == "md5" {
+			h = md5.New()
+			response.HashAlgoUsed = "md5"
+		} else if algo == "sha1" {
+			h = sha1.New()
+			response.HashAlgoUsed = "sha1"
+		} else if algo == "sha224" {
+			h = sha256.New224()
+			response.HashAlgoUsed = "sha224"
+		} else if algo == "sha256" {
+			h = sha256.New()
+			response.HashAlgoUsed = "sha256"
+		} else if algo == "sha384" {
+			h = sha512.New384()
+			response.HashAlgoUsed = "sha384"
+		} else if algo == "sha512" {
+			h = sha512.New()
+			response.HashAlgoUsed = "sha512"
+		}
```

Comment:
And what if `default`?

Ah… you handle it below with the `h == nil`. Wouldn’t it be more clear to put it as an actual case in a `switch`? Then, if we come out of the `switch` we will know that `h` must be valid.